src/Security/samples/StaticFilesAuth/Startup.cs

```diff
@@ -134,12 +123,14 @@ private static void SetFileEndpoint(HttpContext context, PhysicalFileProvider fi
         var fileSystemPath = GetFileSystemPath(files, context.Request.Path);
         if (fileSystemPath != null)
         {
-            var metadata = new List<object>();
-            metadata.Add(new DirectoryInfo(Path.GetDirectoryName(fileSystemPath)));
-            metadata.Add(new AuthorizeAttribute(policy));
+            var metadata = new List<object>
+            {
+                new DirectoryInfo(Path.GetDirectoryName(fileSystemPath)),
+                new AuthorizeAttribute(policy)
+            };
 
             var endpoint = new Endpoint(
-                c => throw new InvalidOperationException("Static file middleware should return file request."),
+                requestDelegate: null,
```

Comment:
This led me to this:

https://github.com/dotnet/aspnetcore/blob/2c557452385bb607da6bc62ba6c14c0a43190bdd/src/Middleware/StaticFiles/src/StaticFileMiddleware.cs#L94-L95

Gross! It's actually pretty clever though. Is this anywhere in our real docs?